simulators/main/shiftplan.py

```diff
@@ -14,16 +14,17 @@
 class Shiftplan:
     polling_interval = one_day
 
-    def __init__(self, shiftplan_model) -> None:
-        self.locationId = ""
-        self.recurringTimeSlots = []
-        self.set_timeslots_for_shiftplan(shiftplan_model)
+    def __init__(self, shiftplan_model, replace_existing_timeslots) -> None:
+        self.locationId = shiftplan_model["locationId"] or ""
+        self.recurringTimeSlots = shiftplan_model["recurringTimeslots"] or []
+        if replace_existing_timeslots.lower() == "true":
+            self.delete_OEE_shiftplan()
         self.add_shiftplan_to_OEE()
 
-    def set_timeslots_for_shiftplan(self, shiftplan):
-        self.locationId = shiftplan["locationId"]
-        self.recurringTimeSlots = shiftplan["recurringTimeslots"]
-
     def add_shiftplan_to_OEE(self):
         oeeAPI.add_timeslots_for_shiftplan(self)
         log.info(f'Added shiftplan to OEE for location: {self.locationId}')
+
+    def delete_OEE_shiftplan(self):
+        oeeAPI.delete_shiftplan(self.locationId)
```

Comment:
I'd prefer to not delete the whole shiftplan but only individual timeslots as suggested yesterday: DELETE /shiftplan/{locationId}/timeslot/{timeslotId}
This "feels" a bit safer b/c we do not delete timeslots that the user created m manually for that location...